src/interfaces/ISavingCircles.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+interface ISavingCircles {
+  struct Circle {
+    address owner;
+    string name;
+    address[] members;
+    uint256 currentIndex;
+    uint256 depositAmount;
+    address token;
+    uint256 depositInterval;
+    uint256 circleStart;
+    uint256 maxDeposits;
+  }
+
+  event CircleCreated(
+    bytes32 indexed id, string name, address[] members, address token, uint256 depositAmount, uint256 depositInterval
+  );
+  event CircleDecommissioned(bytes32 indexed id);
+  event FundsDeposited(bytes32 indexed id, address indexed member, uint256 amount);
+  event FundsWithdrawn(bytes32 indexed id, address indexed member, uint256 amount);
+  event TokenAllowed(address indexed token, bool indexed allowed);
+
+  error AlreadyDeposited();
+  error AlreadyExists();
+  error InvalidDeposit();
+  error InvalidCircle();
+  error NotCommissioned();
+  error NotMember();
+  error NotDecommissionable();
+  error NotWithdrawable();
+  error TransferFailed();
+
+  // External functions (state-changing)
+  function initialize(address owner) external;
```

Comment:
isn't the wonderland convention to put the natspec here and inheritdoc the contract?

Reply:
oh idk, is it? seems sort of strange but if that's convention i can change it

Reply:
fwiw i personally dont really like that convention